File: AmpedBiz/AmpedBiz.Web/src/common/models/supplier.ts
import { Address } from './address';
import { Contact } from './contact';

export type Supplier = {
  id?: string;
  code?: string;
  name?: string;
  contactPerson?: string;
  address?: Address;
  contact?: Contact;
  assigned?: boolean;
}

export type SupplierPageItem = {
  id?: string;
  code?: string;
  name?: string;
  contactPerson?: string;
  address?: Address;
  contact?: Contact;
}

export type SupplierReportPageItem = {
  id?: string;
  code?: string;
  name?: string;
  contactPerson?: string;
  contact?: Contact;
  address?: Address;
}

File: AmpedBiz/AmpedBiz.Service/Dto/Supplier.cs
using System;

namespace AmpedBiz.Service.Dto
{
    public class Supplier
    {
        public Guid Id { get; set; }

        public string Code { get; set; }

        public string Name { get; set; }

        public string ContactPerson { get; set; }

        public Address Address { get; set; }

        public Contact Contact { get; set; }

        public bool Assigned { get; set; }
    }

    public class SupplierPageItem
    {
        public Guid Id { get; set; }

        public string Code { get; set; }

        public string Name { get; set; }

        public string ContactPerson { get; set; }

        public Address Address { get; set; }

        public Contact Contact { get; set; }
    }

    public class SupplierReportPageItem
    {
        public Guid Id { get; set; }

        public string Code { get; set; }

        public string Name { get; set; }

        public string ContactPerson { get; set; }

        public Address Address { get; set; }

        public Contact Contact { get; set; }
    }
}
